charts/rancher-logging/rancher-logging/100.0.0+up3.10.0/templates/_helpers.tpl

```diff
@@ -120,4 +120,29 @@ Set kube-audit file name based on distribution
 {{- else -}}
 {{ required "Filename of the kube-audit log is required" .Values.additionalLoggingSources.kubeAudit.auditFilename }}
 {{- end -}}
+{{- end -}}
+
+{{/*
+A shared list of custom parsers for the vairous fluentbit pods rancher creates
+*/}}
+{{- define "logging-operator.parsers" -}}
+[PARSER]
+    Name              klog
+    Format            regex
+    Regex             ^(?<level>[IWEF])(?<timestamp>\d{4} \d{2}:\d{2}:\d{2}).\d{6} +?(?<thread_id>\d+) (?<filename>.+):(?<linenumber>\d+)] (?<message>.+)
```

Comment:
What does this regex do?

Reply:
it parses klog and extracts the values for each of the capture groups.